mpqp/execution/providers/ibm.py

```diff
@@ -247,23 +225,35 @@ def submit_ibmq(job: Job) -> tuple[str, RuntimeJob | IBMJob]:
     if TYPE_CHECKING:
         assert isinstance(qiskit_circuit, QuantumCircuit)
     qiskit_circuit = qiskit_circuit.reverse_bits()
-
     service = get_QiskitRuntimeService()
-    backend_str = job.device.value
-    session = Session(service=service, backend=backend_str)
+    assert isinstance(job.device, IBMDevice)
+    backend = get_backend(job.device)
+    session = Session(service=service, backend=backend)
+    qiskit_circuit = transpile(qiskit_circuit, backend)
+
     if job.job_type == JobType.OBSERVABLE:
         assert isinstance(job.measure, ExpectationMeasure)
         estimator = Runtime_Estimator(session=session)
         qiskit_observable = job.measure.observable.to_other_language(Language.QISKIT)
-        assert isinstance(qiskit_observable, Operator)
+        assert isinstance(qiskit_observable, SparsePauliOp)
 
-        ibm_job = estimator.run(
-            qiskit_circuit, qiskit_observable, shots=job.measure.shots
+        # Fills the Pauli strings with identities to make the observable size match the circuit size
+        qiskit_observable = SparsePauliOp(
+            [pauli._pauli_list[0].tensor(Pauli("I"*(qiskit_circuit.num_qubits - job.measure.observable.nb_qubits)))
+             for pauli in qiskit_observable],
+            coeffs=qiskit_observable.coeffs
         )
+
+        # FIXME: when we precise the target precision like this, it does not give the right number of shots at the end.
```

Comment:
Try to use the answer provided here https://github.com/Qiskit/qiskit-ibm-runtime/issues/1749

Reply:
does this need an answer ? otherwise you can just resolve the conversation

Reply:
We created a task on Trello for that, this can be done later in another branch.